### ISSUE TITLE
Feature: `sort-objects` Add an option to disable destructured objects sorting

### ISSUE THREAD

**Issue by vladshcherbin** [azat-io/eslint-plugin-perfectionist#392]
### What rule do you want to change?

sort-objects

### Describe the problem

When using `sort-objects` rule with `customGroups`, the rule is also applied to destructured objects. I need custom groups to be used only for objects and no sorting (or also customizable one) for destructured objects.

### Code example

Given the config:

```js
import perfectionist from 'eslint-plugin-perfectionist'

export default {
  plugins: {
    perfectionist
  },
  rules: {
    'perfectionist/sort-objects': ['error', {
      customGroups: {
        id: '^id$'
      },
      groups: [
        'id',
        'unknown'
      ],
      type: 'natural'
    }]
  }
}
```

✅ Object is sorted perfectly fine with `id` being first:

```js
const movie = {
  id: 1,
  createdAt: 2024,
  title: 'movie'
}
```

❌ However, function with destructured object also follows the same rules and will give an error:

```js
// Expected "id" (id) to come before "createdAt" (unknown).
function execute({ createdAt, id, title }) {}
``` 

Same goes for React functional components where using destructured `props` object is a common pattern:

```jsx
// Expected "id" (id) to come before "children" (unknown).
export default function Task({ children, id, title }) {
  return (
    <p>hello darkness my old friend</p>
  )
}
```

### Additional comments

There is an option `destructureOnly` which applies the rule only to destructured objects and I kinda need the opposite.

I've also used [eslint-plugin-sort-destructure-keys](https://github.com/mthadley/eslint-plugin-sort-destructure-keys) for sorting destructured objects so it would be a perfect combo.

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Hello, thank you for your issue.

Could you please tell me why you are using `eslint-plugin-sort-destructure-keys` and what you are missing in `eslint-plugin-perfectionist`?

**Comment by vladshcherbin** [comment]
@azat-io I’ve been using `eslint-plugin-sort-destructure-keys` for at least 4 years, well before `perfectionist`. I’d love to fully transition to `perfectionist`, but I need the ability to sort destructured objects in `natural` order without using my `customGroup` settings.

**Comment by azat-io** [comment]
I see your point. We originally did the `destructureOnly` option because some users wanted to keep the logical key order and not sort objects.

Maybe we should add an option not to use grouping when sorting destructured objects?

**Comment by vladshcherbin** [comment]
@azat-io honestly, I'd be happy with any option or solution giving this result

I was a bit surprised to see the rule apply `customGroups` to my react destructured object `props` and not finding any previous issue about it 😅

**Comment by hugop95** [comment]
@azat-io I think that transforming `destructureOnly` to `allowDestructure` (while keeping the previous one for retro-compatibility) and adding a `fullObject: boolean` (by default `true`) could give the flexibility for users to choose exactly what type of objects they want to sort.

**Comment by azat-io** [comment]
@hugop95 Yes, I like your idea. 👍

**Comment by hugop95** [comment]
@azat-io 🙏 Let me know if you want to handle that or if you're already on it.

**Comment by azat-io** [comment]
@hugop95 I didn't do it, if you want, you can implement it. 🙃

**Comment by vladshcherbin** [comment]
Is there a timeline of a release? I'd absolute love to start using this 🙌

**Comment by azat-io** [comment]
Added in 5606fad3af8c589999357b3f217f55cdb7ec2c29.
Released in [v4.1.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.1.0).
